server/src/main/java/org/elasticsearch/common/geo/GeometryNormalizer.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.geo;
+
+import org.elasticsearch.geometry.Circle;
+import org.elasticsearch.geometry.Geometry;
+import org.elasticsearch.geometry.GeometryCollection;
+import org.elasticsearch.geometry.GeometryVisitor;
+import org.elasticsearch.geometry.Line;
+import org.elasticsearch.geometry.LinearRing;
+import org.elasticsearch.geometry.MultiLine;
+import org.elasticsearch.geometry.MultiPoint;
+import org.elasticsearch.geometry.MultiPolygon;
+import org.elasticsearch.geometry.Point;
+import org.elasticsearch.geometry.Polygon;
+import org.elasticsearch.geometry.Rectangle;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.elasticsearch.common.geo.GeoUtils.normalizePoint;
+
+/**
+ * Transforms provided {@link Geometry} into a lucene friendly format by normalizing latitude and longitude
+ * coordinates and breaking geometries that cross the dateline.
+ */
+public class GeometryNormalizer {
```

Comment:
That looks like an utility class, why not make it `final` as well then?

Reply:
makes sense